src/Storage/Notmuch.hs

```diff
@@ -61,7 +60,5 @@ getFromNotmuchConfig key = do
   stdout <- readProcess cmd args []
   pure $ filter (/= '\n') stdout
 
-isNewMail :: HasTags Message => Message -> T.Text -> IO Bool
-isNewMail m newTag = do
-  t <- tags m
-  pure $ newTag `elem` (decodeUtf8 <$> t)
+isNewMail :: [Tag] -> T.Text -> IO Bool
```

Comment:
Same here; this should be pure, and furthermore should be abstracted over `Foldable t` i.e.

```haskell
isNewMail :: Foldable t => t Tag -> T.Text -> Bool
```

Reply:
Yeah good point actually!